packages/symfony-static-dumper/src/Command/DumpStaticSiteCommand.php

```diff
@@ -21,23 +22,27 @@ public function __construct(
         ParameterBagInterface $parameterBag
     ) {
         parent::__construct();
-
-        $projectDir = (string) $parameterBag->get('kernel.project_dir');
-
-        $this->publicDirectory = $projectDir . '/public';
-        $this->outputDirectory = getcwd() . '/output';
     }
 
     protected function configure(): void
     {
         $this->setName('dump-static-site');
         $this->setDescription('Dump website to static HTML and CSS in the output directory');
+
+        // Adding arguments options for the main command
+        $this->addOption('public-dir', 'p', InputOption::VALUE_REQUIRED, 'Define the input public directory absolute to the root of the project', '/public');
+        $this->addOption('output-dir', 'o', InputOption::VALUE_REQUIRED, 'Define the output directory absolute to the execution of the comand', '/output');
```

Comment:
I meant from the original code:

```php
$projectDir = (string) $parameterBag->get('kernel.project_dir');

        $this->publicDirectory = $projectDir . '/public';
        $this->outputDirectory = getcwd() . '/output';
```

Those should be used as default values:

```php
        // Adding arguments options for the main command
        $this->addOption('... → '/public');
        $this->addOption(... →  '/output');
```

Reply:
The thing is that in the execute function when defining the values it will get the option from public-dir or output-dir and concatenate the value of the project directory or current execution directory to the overall value, so if I add these to the addOption and the user passes the option as an argument, in that case, the user path will be incomplete:

```
symfony console dump-static-site -o /dist -i /static
```
Will generate: /dist as value
while not passing it will get /{project_dir}/public or
/{getcwd()}/output

```php
    protected function execute(InputInterface $input, OutputInterface $output): int
    {   
        $projectDir = (string) $this->parameterBag->get('kernel.project_dir');

        $this->publicDirectory = $projectDir . $input->getOption('public-dir');
        $this->outputDirectory = getcwd() . $input->getOption('output-dir');

        $this->symfonyStyle->section('Dumping static website');
        $this->symfonyStaticDumperApplication->run($this->publicDirectory, $this->outputDirectory, $input->getOption('wdp-only'));

        $this->symfonyStyle->note('Run local server to see the output: "php -S localhost:8001 -t output"');

        return self::SUCCESS;
    }
```

or am I missing something?

Thanks for your time :)
regards,

Reply:
I'm not sure what the issue is, as I'd have to run it locally, so I'm guessing only :)

The correct syntax of custom could be: 

```bash
symfony console dump-static-site -o dist -i static
```

The `/` at start usually goes for absolute value. These values should not be magically composed, but the same as we input them.

Not sure why the `$projectDir` is there actually. It could be `getcwd()` instead:

```php
$this->addOption('public-dir', 'p', InputOption::VALUE_REQUIRED, 'Define the input public directory', getcwd() . '/public');
$this->addOption('output-dir', 'o', InputOption::VALUE_REQUIRED, 'Define the output directory', getcwd() . '/output');
```

Reply:
This is the issue I am finding with that implementation:
```console
vudrok@vwsc:~/Work/stratis/wtw/digicard-api$ symfony console dump-static-site 

/home/vudrok/Work/stratis/wtw/digicard-api/home/vudrok/Work/stratis/wtw/digicard-apipublic
------------------------------------------------------------------------------------------

/home/vudrok/Work/stratis/wtw/digicard-api/home/vudrok/Work/stratis/wtw/digicard-apioutput
------------------------------------------------------------------------------------------

vudrok@vwsc:~/Work/stratis/wtw/digicard-api$ symfony console dump-static-site -o dist -p source-dir

/home/vudrok/Work/stratis/wtw/digicard-apisource-dir
----------------------------------------------------

/home/vudrok/Work/stratis/wtw/digicard-apidist
----------------------------------------------
```
In the first output, it shows how the default value would work, this can be fixed by removing the getcwd() assignation to the variable, but then it will show as follows:
```console
vudrok@vwsc:~/Work/stratis/wtw/digicard-api$ symfony console dump-static-site -o dist -p source-dir

source-dir
----------

dist
----

vudrok@vwsc:~/Work/stratis/wtw/digicard-api$ symfony console dump-static-site

/home/vudrok/Work/stratis/wtw/digicard-apipublic
------------------------------------------------

/home/vudrok/Work/stratis/wtw/digicard-apioutput
------------------------------------------------
```
Some of the basic normative for options from the console are to use path absolute or relative to the current execution directory, so I could do is a little more parsing on that input value is check if it has a starting dot(.) to process it as relative or if not to use it has absolute, as well check for starting / to complete the output value.

Thank you, regards.

Reply:
I added a parser for the path inputs, it will output as follows:

```console
vudrok@vwsc:~/Work/stratis/wtw/digicard-api$ symfony console dump-static-site -o dist -p source-dir

/home/vudrok/Work/stratis/wtw/digicard-api/source-dir
-----------------------------------------------------

/home/vudrok/Work/stratis/wtw/digicard-api/dist
-----------------------------------------------

vudrok@vwsc:~/Work/stratis/wtw/digicard-api$ symfony console dump-static-site

/home/vudrok/Work/stratis/wtw/digicard-api/public
-------------------------------------------------

/home/vudrok/Work/stratis/wtw/digicard-api/output
-------------------------------------------------
```
So running with and without options the output will look correctly setup, what do you think?

This is the code:
```php
    protected function configure(): void
    {
        $this->setName('dump-static-site');
        $this->setDescription('Dump website to static HTML and CSS in the output directory');

        // Adding arguments options for the main command
        $this->addOption('public-dir', 'p', InputOption::VALUE_REQUIRED, 'Define the input public directory absolute to the root of the project', './public');
        $this->addOption('output-dir', 'o', InputOption::VALUE_REQUIRED, 'Define the output directory absolute to the execution of the comand', './output');
    }

    protected function execute(InputInterface $input, OutputInterface $output): int
    {   
        $this->publicDirectory = $this->parseInputPath($input->getOption('public-dir'));
        $this->outputDirectory = $this->parseInputPath($input->getOption('output-dir'));

        $this->symfonyStyle->section('Dumping static website');
        $this->symfonyStaticDumperApplication->run($this->publicDirectory, $this->outputDirectory);

        $this->symfonyStyle->note('Run local server to see the output: "php -S localhost:8001 -t output"');

        return self::SUCCESS;
    }

    protected function parseInputPath($path): string
    {
        $output = '';
        // check if path is absolute will false if relative
        if (strpos($path, '/') === 0) {
            $output = $path;
        }
        // check if path is relative 
        if (strpos($path, '.') === 0) {
            $output = getcwd() . ltrim($path, '.');
        }
        // check if path is neither will default to relative
        if (strlen($output) == 0) {
            $output = getcwd() . '/' . $path;
        }
        return  $output;
    }
```

Reply:
Hi, 

to be honest, I got a bit lost in these :) all I want is to keep the original value as before, passing them as defaults options.
No more modifications or extra code are needed.

Reply:
Working on it :)